cmd/asset-syncer/utils_test.go

```diff
@@ -192,6 +193,19 @@ func Test_syncURLInvalidity(t *testing.T) {
 	}
 }
 
+func Test_getOCIRepo(t *testing.T) {
+	t.Run("it should add the auth header to the resolver", func(t *testing.T) {
+		repo, _ := getOCIRepo("namespace", "test", "https://test", "Basic auth", []string{})
+		// The header property is private so we need to use reflect to get its value
+		resolver := repo.(*OCIRegistry).puller.(*ociPuller).resolver
+		resolverValue := reflect.ValueOf(resolver)
+		headerValue := reflect.Indirect(resolverValue).FieldByName("header")
```

Comment:
yikes, there must be a better way here?

Reply:
It's not clear to my why you can't access `repo.AuthorizationHeader`, since it's a promoted field from the embedded `RepoInternal` struct? For example, this works: https://play.golang.org/p/6jdAMxnRqHZ

Reply:
I can access that field but it would not test the code added in this PR. The goal is to test that the `resolver` has that header.

Reply:
Yeah, OK. There are other ways around it, but probably no prettier. What I've done in similar situations is to pull out the one line which creates the resolver into its own function, say `createResolver(header string)` or similar, which will be the one piece of untested code, then pass the function as an arg to `getOCIRepo`. In tests you can then pass a fake of the same function and just ensure it's called with the header. But yes, it's also not ideal and leaves the one line untested. +1 with what you think best.

Reply:
Yes, also, the `resolver` is already an interface and I have another interface for the `puller`. It's already getting difficult to read with those interfaces, better to skip that one for that line.